exercises/practice/binary-search-tree/BinarySearchTreeTests.cs

```diff
@@ -1,84 +1,199 @@
-using System.Linq;
 using Xunit;
+using System.Text.RegularExpressions;
 
 public class BinarySearchTreeTests
 {
     [Fact]
-    public void Data_is_retained()
+    public void New_bst_is_empty()
     {
-        var tree = new BinarySearchTree(4);
-        Assert.Equal(4, tree.Value);
+        var sut = new BinarySearchTree<int>();
+        Assert.Equal(0, sut.Count);
     }
 
     [Fact(Skip = "Remove this Skip property to run this test")]
-    public void Smaller_number_at_left_node()
+    public void New_bst_can_be_serialized_to_json()
     {
-        var tree = new BinarySearchTree(new[] { 4, 2 });
-        Assert.Equal(4, tree.Value);
-        Assert.Equal(2, tree.Left.Value);
+        var sut = new BinarySearchTree<int>();
```

Comment:
The binary search tree is generic, but we're only using `int` as the type. I think we should also add some non-int type (probably strings).

Reply:
Yes, I proposed it myself higher up in the comments, so definitely agree. But I wanted first to get the shape of the exercise in the right format before I will add more tests and details like that. It's definitely coming.